templates/cli/lib/commands/pull.js.twig

```diff
@@ -26,142 +23,77 @@ const pull = new Command("pull")
-    if (answers.start === "new") {
-        response = await teamsCreate({
-            teamId: 'unique()',
-            name: answers.project,
-            sdk,
-            parseOutput: false
-        })
+    try {
+        let response = await projectsList({
+            parseOutput: false,
+            queries: [JSON.stringify({ method: 'equal', attribute: '$id', values: [localConfig.getProject().projectId] })]
 
-        let teamId = response['$id'];
-        response = await projectsCreate({
-            projectId: answers.id,
-            name: answers.project,
-            teamId,
-            parseOutput: false
         })
-
-        localConfig.setProject(response['$id'], response.name);
-    } else {
-        localConfig.setProject(answers.project.id, answers.project.name);
+        if(response.total === 1){
+            localConfig.setProject(response.projects[0].$id, response.projects[0].name);
+        }
+        success();
+    } catch (e) {
+        throw e;
     }
-    success();
 }
 
-const pullFunction = async () => {
-    // TODO: Add CI/CD support (ID, name, runtime)
-    const answers = await inquirer.prompt(questionsPullFunction)
-    const functionFolder = path.join(process.cwd(), 'functions');
+const pullFunctions = async ({ all, yes } = {}) => {
+    let functions = [];
+    let questions = questionsPullFunctions;
 
-    if (!fs.existsSync(functionFolder)) {
-        fs.mkdirSync(functionFolder, {
-            recursive: true
-        });
+    const localFunctions = localConfig.getFunctions();
+
+    if (all) {
+        questions = yes ? [] : questionsPullFunctions[1];
+        functions = (await paginate(functionsList, { parseOutput: false }, 100, 'functions')).functions;
     }
 
-    const functionId = answers.id === 'unique()' ? ID.unique() : answers.id;
-    const functionDir = path.join(functionFolder, functionId);
+    const answers = await inquirer.prompt(questions);
 
-    if (fs.existsSync(functionDir)) {
-        throw new Error(`( ${functionId} ) already exists in the current directory. Please choose another name.`);
-    }
+    const overridingLocalChanges = yes ?? answers.override.toLowerCase() === "yes";
+    const selectedFunctions = functions.length === 0 ? answers.functions : functions;
 
-    if (!answers.runtime.entrypoint) {
-        log(`Entrypoint for this runtime not found. You will be asked to configure entrypoint when you first push the function.`);
-    }
+    for (let func of selectedFunctions) {
+        const functionExistLocally = localFunctions.find((localFunc) => localFunc['$id'] === func['$id']) !== undefined;
 
-    if (!answers.runtime.commands) {
-        log(`Installation command for this runtime not found. You will be asked to configure the install command when you first push the function.`);
-    }
+        if (!overridingLocalChanges && functionExistLocally) {
+            log(`Skipping locally found implementation of ${func['name']}`)
+            continue;
+        }
```

Comment:
Not sure this makes sense. We dotn have `overridingLocalChanges` in other commands, so I would guess this is saying if source code should be pulled or not.

This `continue` is before pulling info into localConfig. I think it should be after.

Also, copy is a bit inconsistent with Appwrite's terminology. Let's use terms like build, deployment, function settings, ..

Reply:
Better idea, just pull without asking questions - both code and settings